shell/src/interpreter.rs

```diff
@@ -0,0 +1,17 @@
+use async_trait::async_trait;
+use anyhow::Result;
+
+#[async_trait]
+pub trait Interpreter {
+    async fn interpret(&self, code: String) -> Result<String>;
+}
+
+pub struct FakeInterpreter;
+
+#[async_trait]
+impl Interpreter for FakeInterpreter {
+    async fn interpret(&self, code: String) -> Result<String> {
```

Comment:
This interface is too contraining: first of all, interpreter should, I guess take something as a `&str` - then, the result almost certainly will be some error enum that has to be pretty printed, showing location where it failed, coloring etc. I am not sure if we are there yet, but...